### PR TITLE
[Helper Plugin] migrate ReactSelect to TypeScript

### PR DESCRIPTION
**Pull request by JuliusGudni**
### Related PR
#17690

### CODE REVIEW

**Comment by JuliusGudni** [comment]
I noticed that ReactSelect has no storybook file. Shall I take a look at that in this PR or is that stuff for a seperate PR?

**Comment by joshuaellis** [comment]
> I noticed that ReactSelect has no storybook file. Shall I take a look at that in this PR or is that stuff for a seperate PR?

I'd leave it, we're deprecating using this.

File: packages/core/helper-plugin/src/components/ReactSelect.tsx [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+import Select, { ClearIndicatorProps, StylesConfig, components } from 'react-select';
 import styled, { useTheme } from 'styled-components';
 
-const ReactSelect = ({ components, styles, error, ariaErrorMessage, ...props }) => {
+type ReactSelectProps = {
+  components?: object;
+  styles?: object;
+  error?: string;
+  ariaErrorMessage?: string;
+};
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/ReactSelect.tsx:13]: This should be extending the `Props` from `react-select`

File: packages/core/helper-plugin/src/components/ReactSelect.tsx [hunk 2]
```diff
@@ -83,16 +81,18 @@ const CarretBox = styled(IconBox)`
...
   return (
     <CarretBox ref={innerRef} paddingRight={3} {...innerProps}>
       <CarretDown />
     </CarretBox>
   );
 };
 
-const getSelectStyles = (theme, error) => {
+const getSelectStyles = (
+  theme: StrapiTheme,
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/ReactSelect.tsx:93]: This should be `DefaultTheme` from `styled-components`

File: packages/core/helper-plugin/src/components/ReactSelect.tsx [hunk 3]
```diff
@@ -83,16 +81,18 @@ const CarretBox = styled(IconBox)`
 /**
  * These props come from `react-select`.
  */
-// eslint-disable-next-line react/prop-types
-const DropdownIndicator = ({ innerRef, innerProps }) => {
+const DropdownIndicator = ({ innerProps, innerRef }: any) => {
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/ReactSelect.tsx:84]: How come you've not typed this? :)

**Review by joshuaellis — CHANGES_REQUESTED** [review]
See inline comments 👍🏼

File: packages/core/helper-plugin/src/components/ReactSelect.tsx [hunk 4]
```diff
@@ -2,14 +2,29 @@ import React from 'react';
...
+  components,
+  Props as RSProps,
+  DropdownIndicatorProps,
+} from 'react-select';
+import styled, { useTheme, DefaultTheme } from 'styled-components';
+
+type ReactSelectProps = {
+  error?: string;
+  ariaErrorMessage?: string;
+};
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/ReactSelect.tsx:17]:
```suggestion
export interface ReactSelectProps extends RSProps {
  error?: string;
  ariaErrorMessage?: string;
};
```

File: packages/core/helper-plugin/src/components/ReactSelect.tsx [hunk 5]
```diff
@@ -2,14 +2,29 @@ import React from 'react';
...
 
-const ReactSelect = ({ components, styles, error, ariaErrorMessage, ...props }) => {
+const ReactSelect = ({
+  components,
+  styles,
+  error,
+  ariaErrorMessage,
+  ...props
+}: RSProps & ReactSelectProps) => {
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/ReactSelect.tsx:25]:
```suggestion
}: ReactSelectProps) => {
```

**Review by joshuaellis — CHANGES_REQUESTED** [review]
minor tweaks but pretty much there.

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**